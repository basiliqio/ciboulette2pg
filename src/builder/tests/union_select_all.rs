use super::*;

#[test]
fn multi() {
    let mut builder = Ciboulette2PostgresBuilder::new();
    let dest_table = CibouletteTableSettings::new(
        Cow::Borrowed("id"),
        Cow::Borrowed("uuid"),
        Some(Cow::Borrowed("public")),
        Cow::Borrowed("peoples"),
    );

	let from_table = CibouletteTableSettings::new(
        Cow::Borrowed("id"),
        Cow::Borrowed("uuid"),
        Some(Cow::Borrowed("public")),
        Cow::Borrowed("articles"),
    );
	builder.gen_union_select_all(vec![from_table, dest_table].iter()).unwrap();
    let res = builder.build().unwrap();
    insta::assert_debug_snapshot!(res);
}

#[test]
fn single() {
    let mut builder = Ciboulette2PostgresBuilder::new();
    let dest_table = CibouletteTableSettings::new(
        Cow::Borrowed("id"),
        Cow::Borrowed("uuid"),
        Some(Cow::Borrowed("public")),
        Cow::Borrowed("peoples"),
    );
	builder.gen_union_select_all(vec![dest_table].iter()).unwrap();
    let res = builder.build().unwrap();
    insta::assert_debug_snapshot!(res);
}

#[test]
fn no_table() {
    let mut builder = Ciboulette2PostgresBuilder::new();
	builder.gen_union_select_all(vec![].iter()).unwrap();
    let res = builder.build().unwrap();
    insta::assert_debug_snapshot!(res);
}
