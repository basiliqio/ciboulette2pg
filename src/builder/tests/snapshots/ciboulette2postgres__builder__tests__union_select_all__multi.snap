---
source: src/builder/tests/union_select_all.rs
expression: res

---
(
    "SELECT * FROM \"public\".\"articles\" UNION ALL SELECT * FROM \"public\".\"peoples\"",
    Ciboulette2SqlArguments {
        inner: [],
    },
)
